app/models/piece.rb

```diff
@@ -73,34 +72,34 @@ def is_obstructed_horizontally(position_x, position_y)
     end
   end
 
-  # 5. See if there is a vertical obstruction
-  def is_obstructed_diagonally(position_x, position_y)
-    x_current = piece.position_x
-    y_current = piece.position_y
-    x_destination = position_x
-    y_destination = position_y
-
-    if x_current < x_destination && y_current < y_destination # up-right diagonal
-      while x_current < x_destination && y_current < y_destination do |x, y|
-        return true if is_occupied?(x += 1)(y += 1) == true
-        end
-      end
-    elsif x_current > x_destination && y_current < y_destination # up-left diagonal
-      while x_current > x_destination && y_current < y_destination do |x, y|
-        return true if is_occupied?(x -= 1)(y += 1) == true
-        end
-      end
-    elsif x_current < x_destination && y_current > y_destination # down-right diagonal
-      while x_current > x_destination && y_current < y_destination do |x, y|
-        return true if is_occupied?(x += 1)(y -= 1) == true
-        end
-      end
-    else
-      while x_current > x_destination && y_current < y_destination do |x, y|
-        return true if is_occupied?(x -= 1)(y -= 1) == true
-      end
-    end
-  end
+  # # 5. See if there is a vertical obstruction
```

Comment:
Why is this commented out? 🤔

Reply:
@ilya It still needs some refactoring as it produces errors.

Reply:
yeah, just saw the issue :)